client/tailscale/localclient.go

```diff
@@ -859,6 +859,34 @@ func (lc *LocalClient) NetworkLockSign(ctx context.Context, nodeKey key.NodePubl
 	return nil
 }
 
+func (lc *LocalClient) AddServeConfig(ctx context.Context, config ipn.ServeConfig) (*ipn.ServeConfigView, error) {
+	var b bytes.Buffer
+	if err := json.NewEncoder(&b).Encode(config); err != nil {
+		return nil, fmt.Errorf("encoding config: %v", err)
+	}
+	body, err := lc.send(ctx, "POST", "/localapi/v0/serve-config", 200, &b)
+	if err != nil {
+		return nil, fmt.Errorf("sending serve config: %w", err)
+	}
+	sc := new(ipn.ServeConfigView)
+	if err := json.Unmarshal(body, sc); err != nil {
+		return nil, fmt.Errorf("unmarshalling ServeConfigView: %w", err)
+	}
+	return sc, nil
```

Comment:
wait, why is this returning the thing it just passed in? Just return an error only?